File: python/grader2.py
print("---- NESTED SCORE GRADER IN PYTHON TERMINAL ----\n")

while True:
    score = float(input("Enter your score between 0 to 100: "))
    
    if (score > 100 or score < 0):
        print("Score is out of range \nTerminating...")
        break
    else:
        if score >= 90:
            print("GRADE: A\n")
        elif score >= 80:
            print("GRADE: B\n")
        elif score >= 70:
            print("GRADE: C\n")
        elif score >= 60:
            print("GRADE: D\n")
        else:
            print("GRADE: F\n")

File: php/grader2.php
<?php

while (true) {
    
    echo "Enter a score between 0 and 100: ";
    $score = trim(fgets(STDIN));

    if (!is_numeric($score)) {
        echo "Error: Enter a valid score.\nTerminating...";
        break;
    }
    else if ($score < 0 || $score > 100) {
        echo "Error: Score is out of range.\nTerminating...";
        break;
    } else {
        if ($score >= 90) {
            echo "Grade: A\n";
        } elseif ($score >= 80) {
            echo "Grade: B\n";
        } elseif ($score >= 70) {
            echo "Grade: C\n";
        } elseif ($score >= 60) {
            echo "Grade: D\n";
        } else {
            echo "Grade: F\n";
        }
    }
}

?>